server/pipeline/create.go

```diff
@@ -88,7 +88,7 @@ func Create(ctx context.Context, _store store.Store, repo *model.Repo, pipeline
 		return nil, ErrFiltered
 	} else if configFetchErr != nil {
 		log.Error().Str("repo", repo.FullName).Err(configFetchErr).Msgf("error while fetching config '%s' in '%s' with user: '%s'", repo.Config, pipeline.Ref, repoUser.Login)
-		return nil, updatePipelineWithErr(ctx, _forge, _store, pipeline, repo, repoUser, fmt.Errorf("could not load config from forge: %w", err))
+		return nil, updatePipelineWithErr(ctx, _forge, _store, pipeline, repo, repoUser, fmt.Errorf("could not load config from forge: %s", err.Error()))
```

Comment:
```diff
-		return nil, updatePipelineWithErr(ctx, _forge, _store, pipeline, repo, repoUser, fmt.Errorf("could not load config from forge: %s", err.Error()))
+		return nil, updatePipelineWithErr(ctx, _forge, _store, pipeline, repo, repoUser, fmt.Errorf("could not load config from forge (%w): %s", err, err.Error()))
```

as now it is redable but we can not pass up errors to be asserted/catched specialy ahndled etc!!!

Reply:
normally %w should be equal %s but allow to get the error back if needed or assert if it is the underlaying.
it it does not format nicely this is the better way to do to not lose that bit of information

Reply:
https://github.com/woodpecker-ci/woodpecker/pull/5518

Reply:
a unit test would not catch those issues.

full integration test suite will, but i paused my work on that :/

Reply:
Don't understand your comments. Have you read the linked issue? As you can see, the error message is not properly formatted at all and therefor error information got lost. Yes, wrapping errors is better to be able to assert some error types, but what's your suggestion to keep error wrapping **and** fix the linked issue where the error message was not shown at all?

Reply:
PS: yes you probably could write an unit test but that's just for this codeblock and it can just be done wront at all other places at another time -> thats why reviews are here :D

i dont think it is an hard issue and i did not test if we use some errors.Is or errors.As up in the caller stack, but its more a contract that somebody might at some point do so - e.g. we should expect golangs error handling system to work